cmd/notation/cert/add.go

```diff
@@ -0,0 +1,81 @@
+package cert
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/notaryproject/notation/cmd/notation/internal/truststore"
+	"github.com/spf13/cobra"
+)
+
+type certAddOpts struct {
+	storeType  string
+	namedStore string
+	path       []string
+}
+
+func certAddCommand(opts *certAddOpts) *cobra.Command {
+	if opts == nil {
+		opts = &certAddOpts{}
+	}
+	command := &cobra.Command{
+		Use:   "add --type <type> --store <name> [flags] <cert_path>...",
+		Short: "Add certificates to the trust store.",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("missing certificate path")
+			}
+			opts.path = args
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return addCerts(opts)
+		},
+	}
+	command.Flags().StringVarP(&opts.storeType, "type", "t", "", "specify trust store type, options: ca, signingAuthority")
+	command.Flags().StringVarP(&opts.namedStore, "store", "s", "", "specify named store")
+	return command
+}
+
+func addCerts(opts *certAddOpts) error {
+	storeType := opts.storeType
+	if storeType == "" {
+		return errors.New("store type cannot be empty")
+	}
+	if !truststore.ValidateStoreType(storeType) {
+		return fmt.Errorf("unsupported store type: %s", storeType)
+	}
+	namedStore := opts.namedStore
+	if namedStore == "" {
+		return errors.New("named store cannot be empty")
+	}
+	var success []string
+	var failure []string
+	var errorSlice []error
+	for _, p := range opts.path {
+		err := truststore.AddCert(p, storeType, namedStore, false)
+		if err != nil {
+			failure = append(failure, p)
+			errorSlice = append(errorSlice, err)
+		} else {
+			success = append(success, p)
+		}
+	}
+
+	//write out
+	if len(success) != 0 {
+		fmt.Printf("Successfully added following certificates to named store %s of type %s:\n", namedStore, storeType)
+		for _, p := range success {
+			fmt.Println(p)
+		}
+	}
+	if len(failure) != 0 {
+		fmt.Printf("Failed to add following certificates to named store %s of type %s:\n", namedStore, storeType)
+
+		for ind := range failure {
+			fmt.Printf("%s, with error %q\n", failure[ind], errorSlice[ind])
+		}
+	}
+
+	return nil
```

Comment:
In case where we failed to add certificate, we should return non-zero exit code.  Applicable to multiple CLIs

Reply:
> In case where we failed to add certificate, we should return non-zero exit code. Comment applicable to all cert commands

@priteshbandi Could you provide an example or a suggestion of change here?

Reply:
For example, user tries to add multiple certificates to trust store using the ` notation certificate add --type ca --store some-store /temp/present.crt /temp/absent.crt` command and one of the certificate(`absent.crt`) doesn't exist 

Since the command will fail to add `absent.crt`, the exit code for the command should be non-zero. Also, the error messages related to failed portion of the command i.e. addition of  `absent.crt` to trust store should be published stderr instead of stdin

Reply:
Sure. Updated the code.

Reply:
Looks like now we are emitting error messages twice:

https://ctxt.io/2/AAAQ8Ra_Ew

Reply:
> Looks like now we are emitting error messages twice:
> 
> https://ctxt.io/2/AAAQ8Ra_Ew

Nice catch! Code updated.